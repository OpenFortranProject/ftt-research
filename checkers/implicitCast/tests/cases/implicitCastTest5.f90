function f()
  real :: f
  real :: x = 42.4

  f = x
end function

program test
  interface
    function f()
    real :: f
    end function
  end interface

  INTEGER :: x
  
  x = f() + 4
  
  print *, x

end
