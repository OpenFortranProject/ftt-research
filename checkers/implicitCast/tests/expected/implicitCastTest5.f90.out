Warning: environment variable COMPASS_PARAMETERS is not set, trying to use compass_parameters of the current path
 Running Prerequisite SgProject
ImplicitCast: /home/dagit/research/checkers/implicitCast/tests/cases/implicitCastTest5.f90:21.2: Implicit cast in Fortran code found:  expr 'f() + 4' from real to integer
